eslint/babel-eslint-plugin/src/rules/no-invalid-this.js

```diff
@@ -1,24 +1,74 @@
-import ruleComposer from "eslint-rule-composer";
 import eslint from "eslint";
+import ruleExtender from "../utils/rule-extender";
 
 const noInvalidThisRule = new eslint.Linter().getRules().get("no-invalid-this");
 
-export default ruleComposer.filterReports(noInvalidThisRule, problem => {
-  let inClassMember = false;
-  let node = problem.node;
-
-  while (node) {
-    if (
-      node.type === "ClassPrivateMethod" ||
-      node.type === "ClassPrivateProperty" ||
-      node.type === "ClassProperty"
-    ) {
-      inClassMember = true;
-      return;
+function isClassMember(node) {
+  return (
+    node.type === "ClassPrivateMethod" ||
+    node.type === "ClassPrivateProperty" ||
+    node.type === "ClassProperty"
+  );
+}
+
+export default ruleExtender(noInvalidThisRule, {
+  metaOverrides: {
+    docs: {
+      url:
+        "https://github.com/babel/babel/tree/main/eslint/babel-eslint-plugin",
+    },
+  },
+  createAdditionalVisitors(context) {
+    // The strict check in the original rule relies on ClassBody
+    // nodes being in strict mode. When the whole module isn't
+    // in strict mode, we need to do an extra check here.
+    if (context.parserOptions.sourceType === "script") {
+      return {
+        "ClassProperty[computed=true] ThisExpression"(node) {
+          let nodeToCheck = node;
+
+          while (nodeToCheck) {
+            if (nodeToCheck.type === "ClassProperty") {
```

Comment:
Tests were passing without this check but we might as well short-circuit the check here for performance reasons.